cedar-policy-core/src/est/expr.rs

```diff
@@ -1096,40 +1150,40 @@ fn interpret_primary(p: cst::Primary) -> Result<Either<ast::Name, Expr>, ParseEr
                         .map(|ASTNode { node, .. }| {
                             node.as_ref()
                                 .ok_or_else(|| {
-                                    ParseErrors(vec![ParseError::ToAST(
+                                    ParseErrors(vec![ParseError::InternalInvariantViolation(
                                         "node should not be empty".to_string(),
                                     )])
                                 })
                                 .and_then(|id| id.to_string().parse().map_err(Into::into))
                         })
                         .collect::<Result<Vec<ast::Id>, _>>()?,
                 ))),
-                (path, Some(id)) => Err(ParseError::ToAST(format!(
+                (path, Some(id)) => Err(ParseError::InvalidExpr(format!(
                     "{} is not a valid expr",
```

Comment:
Drop the "is not a valid expr" + same comment for the other instances in this file